test/SILGen/tsan_instrumentation.swift

```diff
@@ -63,3 +64,53 @@ func inoutGlobalClassStoredProperty() {
   // temporary buffer passed to takesInout().
   takesInout(&gClass.storedProperty)
 }
+
+// Known-empty types don't have storage, so there is no address
+// to pass to the TSan runtime to check for data races on inout accesses.
+// In this case, the instrumentation should skip the call to
+// __tsan_external_write()
+
+struct ZeroSizedStruct {
+  mutating
```

Comment:
Tiny nit: putting the `mutating` on its own line isn't the convention we use in the Swift parts of the repo.